src/blockchain.js

```diff
@@ -29,10 +29,16 @@ Blockchain.connect = function(connectionList, opts, doneCb) {
 
   const checkConnect = (next) => {
     this.blockchainConnector.getAccounts((err, _a) => {
-      if (err) {
-        this.blockchainConnector.setProvider(null);
-      }
-      return next(null, !err);
+      const provider = self.blockchainConnector.getCurrentProvider();
+      const connString = provider.host;
+
+      if (err) this.blockchainConnector.setProvider(null);
```

Comment:
That can be first no?
Before we request `provider` and `connString`

Reply:
If I don't request the provider first it gets lost on an error situation.

Or maybe I'm not understanding your comment.

Reply:
yep you're right, by trying to code what I meant, I realize you can ignore what I said